code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -442,6 +442,8 @@
 				fart_type = 2
 			else if(prob(12)) // 0.4%
 				fart_type = 3
+				if(loc.z != 1)
```

Comment:
You shouldn't take the fact that loc exists as obvious, it could be null aswell. Find another way to check the zlevel, such as var/turf/T = get_turf(src); if(T && T.z != 1) doshit()